File: Selenium/CSharp/android/PerfectoCSharpAndroidSample/Sample.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Remote;
using OpenQA.Selenium.Support.UI;
using Reportium.Client;
using Reportium.Model;
using Reportium.Test.Result;
namespace PerfectoCSharpAndroidSample
{
    [TestClass]
    public class Sample
    {
        [TestMethod]
        public void AndroidTest()
        {
			DesiredCapabilities capabilities = new DesiredCapabilities();

			// 1. Replace <<cloud name>> with your perfecto cloud name (e.g. demo is the cloudName of demo.perfectomobile.com).
			String cloudName = "<<cloud name>>";

			// 2. Replace <<security token>> with your perfecto security token.
			String securityToken = "<<security token>>";
			capabilities.SetCapability("securityToken", securityToken);

			// 3. Set web capabilities.
			capabilities.SetCapability("platformName", "Android");
			capabilities.SetCapability("model", "Galaxy S.*");
			capabilities.SetCapability("browserName", "Chrome");
			capabilities.SetCapability("useAppiumForWeb", true);
            capabilities.SetCapability("openDeviceTimeout", 5);


			// Initialize the  driver
			RemoteWebDriver driver = new RemoteWebDriver(
					new Uri("https://" + cloudName.Replace(".perfectomobile.com", "")
					+ ".perfectomobile.com/nexperience/perfectomobile/wd/hub"),
					capabilities);

			// Setting implicit wait
			driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(5);

			PerfectoExecutionContext perfectoExecutionContext;
			if (System.Environment.GetEnvironmentVariable("jobName") != null)
			{
				perfectoExecutionContext = new PerfectoExecutionContext.PerfectoExecutionContextBuilder()
						.WithProject(new Project("My Project", "1.0"))
						.WithJob(new Job(System.Environment.GetEnvironmentVariable("jobName"),
								int.Parse(System.Environment.GetEnvironmentVariable("jobNumber"))))
						.WithWebDriver(driver).Build();
			}
			else
			{
				perfectoExecutionContext = new PerfectoExecutionContext.PerfectoExecutionContextBuilder()
						.WithProject(new Project("My Project", "1.0"))
						.WithWebDriver(driver).Build();
			}
			ReportiumClient reportiumClient = PerfectoClientFactory.CreatePerfectoReportiumClient(perfectoExecutionContext);

			reportiumClient.TestStart("Selenium C# Android Sample", new Reportium.Test.TestContext("selenium", "android"));

			try
			{
				WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(30));
				String search = "perfectomobile";
				reportiumClient.StepStart("Navigate to Google");
				Dictionary<String, Object> pars = new Dictionary<String, Object>();
				driver.Url = "https://www.google.com";
				reportiumClient.StepEnd();

				reportiumClient.StepStart("Search for Perfecto");
				IWebElement searchbox = wait.Until(ExpectedConditions.ElementToBeClickable(driver.FindElement(By.XPath("//*[@name='q']"))));
				searchbox.SendKeys(search);
				searchbox.SendKeys(Keys.Enter);
				reportiumClient.StepEnd();

				reportiumClient.StepStart("Verify Title");
				String expectedText = "perfectomobile - Google Search";
				Console.WriteLine(driver.Title);
				// Adding assertions to the Execution Report. This method will not fail the test
				reportiumClient.ReportiumAssert(expectedText, driver.Title.Contains(expectedText));
				reportiumClient.StepEnd();

				reportiumClient.TestStop(TestResultFactory.CreateSuccess());
			}
			catch (Exception e)
			{
				reportiumClient.TestStop(TestResultFactory.CreateFailure(e));
			}

			// Prints the Smart Reporting URL
			String reportURL = reportiumClient.GetReportUrl();
			Console.WriteLine("Report url - " + reportURL);

			// Quits the driver
			driver.Quit();
		}
    }
}

File: Selenium/Ruby/android/sample.rb
require 'perfecto-reporting'
require 'appium_lib'
require 'selenium-webdriver'

desired_caps = {
        #  1. Replace <<cloud name>> with your perfecto cloud name (e.g. demo is the cloudName of demo.perfectomobile.com).
    appium_lib: {
        server_url: 'https://%s.perfectomobile.com/nexperience/perfectomobile/wd/hub' % "<<cloud name>>",  
    },
    caps: {
        #  2. Replace <<security token>> with your perfecto security token.
        securityToken: '<<security token>>',
        
        # 3. Set device capabilities.
        platformName: 'Android',
        model: 'Galaxy S.*',

        # Set other capabilities.
        browserName: 'mobileOS',
        useAppiumForWeb: true,
        openDeviceTimeout: 5
    }
}
# Initialize the Appium driver
@driver = Appium::Driver.new(desired_caps, true).start_driver

# Setting implicit wait
@driver.manage.timeouts.implicit_wait = 5

# Initialize Smart Reporting
if ENV["jobName"] != nil
    perfectoExecutionContext = PerfectoExecutionContext.new(PerfectoExecutionContext::PerfectoExecutionContextBuilder
    .withWebDriver(@driver).withJob(Job.new(ENV["jobName"], ENV["jobNumber"].to_i)).build)
else
    perfectoExecutionContext = PerfectoExecutionContext.new(PerfectoExecutionContext::PerfectoExecutionContextBuilder
            .withWebDriver(@driver).build)
end
@reportiumClient = PerfectoReportiumClient.new(perfectoExecutionContext)
tec = TestContext::TestContextBuilder.build()
@reportiumClient.testStart("Selenium Ruby Android Sample", tec)

begin
    timeout = 30
    wait = Selenium::WebDriver::Wait.new(:timeout => timeout)
    search = "perfectomobile"
    @reportiumClient.stepStart('Navigate to Google');
    @driver.get('https://www.google.com');
    @reportiumClient.stepEnd();

    @reportiumClient.stepStart('Search for ' + search);
    wait.until{ @driver.find_element(:name => 'q') }
    @driver.find_element(:name => 'q').send_keys(search)
    @driver.find_element(:name => 'q').send_keys:return
    @reportiumClient.stepEnd();

    @reportiumClient.stepStart('Verify Title');
    expectedText = "perfectomobile - Google Search";
    @reportiumClient.reportiumAssert(expectedText, @driver.title === expectedText)
    @reportiumClient.stepEnd();
            
    @reportiumClient.testStop(TestResultFactory.createSuccess(), tec)

rescue Exception => exception
    @exception = exception
    @reportiumClient.testStop(TestResultFactory.createFailure(@exception.exception.message, @exception.exception, nil), tec)
    raise exception
ensure
    # Prints the report url
     puts 'Report url - ' + @reportiumClient.getReportUrl
     
    #Quits the driver
    @driver.quit
    puts "Ruby Android Execution completed"
end
